tests/integration/cluster-users/sar-test.sh

```diff
@@ -240,6 +277,92 @@ function runTests() {
 
 	echo "--> ${VIEW_EMAIL} should NOT be able to create ory Access Rule"
 	testPermissions "create" "rule.oathkeeper.ory.sh" "${NAMESPACE}" "no"
+
+	EMAIL=${NAMESPACE_ADMIN_EMAIL} PASSWORD=${NAMESPACE_ADMIN_PASSWORD} getConfigFile
+	export KUBECONFIG="${PWD}/kubeconfig"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create new namespace"
+	createNamespaceForNamespaceAdmin
+	export SHOULD_CLEANUP_NAMESPACE="true"
+
+	# namespace admin should not be able to get or create any resource in system namespaces
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should NOT be able to list Deployments in system namespace"
+	testPermissions "list" "deployment" "${SYSTEM_NAMESPACE}" "no"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should NOT be able to get ory Access Rule in system namespace"
+	testPermissions "get" "rule.oathkeeper.ory.sh" "${SYSTEM_NAMESPACE}" "no"
+
+	echo "--> ${VIEW_EMAIL} should NOT be able to create secret in system namespace"
+	testPermissions "create" "secret" "${SYSTEM_NAMESPACE}" "no"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should NOT be able to delete system namespace"
+	testPermissions "delete" "namespace" "${SYSTEM_NAMESPACE}" "no"
+
+	# namespace admin should be able to get/create k8s and kyma resources in the namespace he created
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to list Deployments in the namespace he created"
+	testPermissions "list" "deployment" "${NAMESPACE_ADMIN_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to get Pods in the namespace he created"
+	testPermissions "get" "pod" "${NAMESPACE_ADMIN_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create ory Access Rule in the namespace he created"
+	testPermissions "create" "rule.oathkeeper.ory.sh" "${NAMESPACE_ADMIN_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create secret in the namespace he created"
+	testPermissions "create" "secret" "${NAMESPACE_ADMIN_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to delete namespace he created"
+	testPermissions "delete" "namespace" "${NAMESPACE_ADMIN_NAMESPACE}" "yes"
+
+	echo "--> ${NAMESPACE_ADMIN_EMAIL} should be able to create clusterrolebindings"
```

Comment:
You mean namespace Admins can bind their own groups to a ClusterRole, say `Kyma-admin`? I'm not sure if this is what we are looking for.

Reply:
You are right. This assertion is wrong. 

This assertion is working only because `kubectl auth can-i` is not smart enough. Look at [this comment](https://github.com/kubernetes/kubernetes/issues/75970#issuecomment-478711115)

> can-i asks the authorization system if the given verb can be performed on the given noun at the given scope. It does not interpret whether the resource exists at that scope.

And a fragment of [this post](https://kubernetes.io/blog/2017/10/using-rbac-generally-available-18/):

> Roles cannot grant access to non-namespaced resources (such as nodes)

The `kyma-admin` ClusterRole which we are binding to `namespace.admin@kyma.cx` in a specific namespace with the RoleBinding grants they access to create ClusterRole and ClusterRoleBinding **in that namespace**. `kubectl auth can-i` then checks if "given verb (create) can be performed on the given noun (ClusterRoleBinding) at the given scope (custom namespace)" and returns true, even though it is actually not possible, as the `ClusterRoleBinding` is a non-namespaced resource.

Reply:
Namespace Admins can bind any groups to a ClusterRole, but only with a RoleBinding, not with a ClusterRoleBinding. They also can't create a ClusterRole.

Reply:
I moved the assertion to test `create clusterrolebindings` in system namespace - because RBAC is permissive and user can create clusterrolebindings only if sar-check returns true for every namespace. Because the user has admin rights only in the "custom" namespace, it won't be able to create it in system namespace.